### ISSUE TITLE
Bug in 1.34.0 when using `timeMin` set to `$now` when getting all events in Google Calendar node

### ISSUE THREAD

**Issue by blastik** [n8n-io/n8n#8961]
### Bug Description

Im getting the following error
```
{
  "errorMessage": "date.endsWith is not a function",
  "errorDetails": {},
  "n8nDetails": {
    "n8nVersion": "1.34.0 (Self Hosted)",
    "binaryDataMode": "default",
    "stackTrace": [
      "TypeError: date.endsWith is not a function",
      "    at hasTimezone (/usr/local/lib/node_modules/n8n/node_modules/n8n-nodes-base/dist/nodes/Google/Calendar/GenericFunctions.js:131:36)",
      "    at addTimezoneToDate (/usr/local/lib/node_modules/n8n/node_modules/n8n-nodes-base/dist/nodes/Google/Calendar/GenericFunctions.js:133:9)",
      "    at Object.execute (/usr/local/lib/node_modules/n8n/node_modules/n8n-nodes-base/dist/nodes/Google/Calendar/GoogleCalendar.node.js:340:83)",
      "    at Workflow.runNode (/usr/local/lib/node_modules/n8n/node_modules/n8n-workflow/dist/Workflow.js:724:42)",
      "    at /usr/local/lib/node_modules/n8n/node_modules/n8n-core/dist/WorkflowExecute.js:660:68",
      "    at processTicksAndRejections (node:internal/process/task_queues:95:5)",
      "    at /usr/local/lib/node_modules/n8n/node_modules/n8n-core/dist/WorkflowExecute.js:1062:20"
    ]
  }
}
```

### To Reproduce

It seems that from 1.34.0 when using `timeMin` parameter set to `$now` it does not work any longer

```
    {
      "parameters": {
        "operation": "getAll",
        "calendar": {
          "__rl": true,
          "value": "whatever@whatever.es",
          "mode": "list",
          "cachedResultName": "whatever@whatever.es"
        },
        "limit": 8,
        "options": {
          "timeMin": "={{ $now }}",
          "orderBy": "startTime",
          "singleEvents": true
        }
      },
      "id": "UID",
      "name": "Google Calendar",
      "type": "n8n-nodes-base.googleCalendar",
      "typeVersion": 1,
      "position": [
        160,
        180
      ],
      "credentials": {
        "googleCalendarOAuth2Api": {
          "id": "UID",
          "name": "whatever"
        }
      }
    },
```

### Expected behavior

Fetch calendar events

### Operating System

Truenas / K8S

### n8n Version

1.34.0

### Node.js Version

18.19.1

### Database

PostgreSQL

### Execution mode

main (default)

**Comment by Joffcom** [comment]
Hey @blastik,

Is this still an issue for you in 1.34.1?

**Comment by blastik** [comment]
fixed in 1.34.1
